src/cargo/ops/fix.rs

```diff
@@ -252,6 +252,59 @@ fn check_version_control(gctx: &GlobalContext, opts: &FixOptions) -> CargoResult
 }
 
 fn migrate_manifests(ws: &Workspace<'_>, pkgs: &[&Package]) -> CargoResult<()> {
+    // HACK: Duplicate workspace migration logic between virtual manifests and real manifests to
+    // reduce duplicate messages being reported to the user
```

Comment:
I'm not sure what kind of duplication message you're referring to here.

Reply:
I've made the comment more explicit